src/actions/sources/select.js

```diff
@@ -137,6 +141,19 @@ export function selectLocation(location: Location) {
 
     dispatch(setSymbols(loadedSource.id));
     dispatch(setOutOfScopeLocations());
+
+    // If this is a new location and there is an active file search open,
+    // then update the results
+    const isLocationChanged =
+      !previousLocation || previousLocation.sourceId !== location.sourceId;
+    if (isLocationChanged) {
+      const isFileSearchOpen = getActiveSearch(getState()) === "file";
+      const fileSearchQuery = getFileSearchQuery(getState());
+      if (isFileSearchOpen && fileSearchQuery) {
+        const editor = getEditor();
+        dispatch(searchContents(fileSearchQuery, editor));
+      }
+    }
```

Comment:
lets try and simplify this a bit...

```js
const newSource = getSelectedSource(getState());
if (currentSource != newSource) {
  dispatch(clearFileSearch())
}
```

we can do a couple of things in `clearFileSearch`

1. check if there is an active file search
2. dispatch `searchContents`